source/Sashimi.Tests.Shared/Server/TestCalamariCommandBuilder.cs

```diff
@@ -216,6 +253,102 @@ IActionHandlerResult ExecuteActionHandler(List<string> args)
                 serverInMemoryLog.ToString());
         }
 
+        IActionHandlerResult ExecuteActionHandlerOutProc(List<string> args)
+        {
+            using (var variablesFile = new TemporaryFile(Path.GetTempFileName()))
+            {
+                variables.Save(variablesFile.FilePath);
+
+                var calamariFullPath = GetOutProcCalamariExePath();
+                Console.WriteLine("Running Calamari OutProc from: "+ calamariFullPath);
+
+                ExecutableHelper.AddExecutePermission(calamariFullPath);
+
+                var commandLine = new CommandLine(calamariFullPath);
+                foreach (var argument in args)
+                    commandLine = commandLine.Argument(argument);
+
+                var calamariResult = Invoke(commandLine);
+
+                var capturedOutput = calamariResult.CapturedOutput;
+
+                var serverInMemoryLog = new ServerInMemoryLog();
+
+                var outputFilter = new ScriptOutputFilter(serverInMemoryLog);
+                foreach (var text in capturedOutput.Errors)
+                {
+                    outputFilter.Write(ProcessOutputSource.StdErr, text);
+                }
+
+                foreach (var text in capturedOutput.Infos)
+                {
+                    outputFilter.Write(ProcessOutputSource.StdOut, text);
+                }
+
+                return new TestActionHandlerResult(calamariResult.ExitCode,
+                    outputFilter.TestOutputVariables, outputFilter.Actions,
+                    outputFilter.ServiceMessages, outputFilter.ResultMessage, outputFilter.Artifacts,
+                    serverInMemoryLog.ToString());
+            }
+        }
+
+        string GetOutProcCalamariExePath()
+        {
+            var calamariFlavour = typeof(TCalamariProgram).Assembly.GetName().Name;
+            var sashimiTestFolder = Path.GetDirectoryName(typeof(TCalamariProgram).Assembly.FullLocalPath());
+
+            if (TestEnvironment.IsCI)
+            {
+                //This is where Teamcity puts the Calamari binaries
+                var calamaribinariesFolder = Environment.GetEnvironmentVariable(CalamaribinariesLocationEnvironmentVariable);
+                return AddExeExtensionIfNecessary(Path.GetFullPath(Path.Combine(sashimiTestFolder, calamaribinariesFolder, calamariFlavour)));
+            }
+
+            //Running locally - change these to your liking
+            var configuration = "Debug"; //Debug;Release
```

Comment:
How would you override? Could we use environment variables like elsewhere?

Reply:
You'd need to modify the code and run it. Would environment variables make it more discoverable? This is only used for running locally